### PR TITLE
Feature/python3 support

### PR DESCRIPTION
**Pull request by robszewczyk**

### CODE REVIEW

File: codegen/weave-device-cpp/templates/common.macros [hunk 1]
```diff
@@ -53,7 +53,7 @@ enum {{ enum.base_name }} {
               chain(get_struct_dependencies(obj)|reject('common')|map(attribute='parent'))|
               chain(get_struct_dependencies(obj)|select('common'))|
               reject('false')|unique|reject('standard')|reject('vendor')|
-              reject('equalto',obj)|sort -%}
+              reject('equalto',obj)|sort(attribute='full_name') -%}
```
Comment (gerickson) [inline codegen/weave-device-cpp/templates/common.macros:56]: This seems unrelated to the Python 3 upgrade. Is it?

Comment (robszewczyk) [reply codegen/weave-device-cpp/templates/common.macros:56]: It is VERY much related.  In Python2, comparison between unrelated objects would yield a stable result, in Python3, in the absence of a comparison operator, this throws an error.  In this case, the objects sorted were derived from a common ancestor, but defined no comparison.  When we were comparing Typespace and Trait, the comparison would throw an error.   I would note that the new code parallels the structure we've seen in other templates for codegen that have not been opensourced yet.

File: configure [hunk 2]
```diff
@@ -3288,7 +3300,7 @@ if ${nl_cv_protoc_version+:} false; then :
   $as_echo_n "(cached) " >&6
 else
 
-		       nl_cv_protoc_version=`${PROTOC} --version 2>&1 | ${SED} -n -e 's/^.\{1,\}\([0-9]\{1,\}.[0-9]\{1,\}.[0-9]\{1,\}\)$/\1/gp'`
+		       nl_cv_protoc_version=`${PROTOC} --version 2>&1 | ${SED} -n -E -e 's/^[^0-9]*(([0-9]+\.)*[0-9]+).*/\1/p'`
```
Comment (mrjerryjohns) [inline configure:3303]: @gerickson you cool with this regex change?

Comment (gerickson) [reply configure:3303]: This should be auto-generated and should match what came in via Eli Ribble from commit ea8339e

Comment (robszewczyk) [reply configure:3303]: Note: this regex change has been previously approved and pulled into `configure.ac` in https://github.com/openweave/openweave-wdlc/pull/39 ; in this pull request we simply update the automake-generated files. The change was made to support protoc-3.6.1 and similar style of versions
